internal/app/machined/pkg/runtime/v1alpha1/platform/nocloud/metadata.go

```diff
@@ -178,14 +179,15 @@ func (n *Nocloud) configFromCD() (metaConfig []byte, networkConfig []byte, machi
 }
 
 //nolint:gocyclo
-func (n *Nocloud) acquireConfig(ctx context.Context) (metadataConfigDl, metadataNetworkConfigDl, machineConfigDl []byte, hostname string, err error) {
+func (n *Nocloud) acquireConfig(ctx context.Context) (metadataConfigDl, metadataNetworkConfigDl, machineConfigDl []byte, err error) {
 	s, err := smbios.GetSMBIOSInfo()
 	if err != nil {
-		return nil, nil, nil, "", err
+		return nil, nil, nil, err
 	}
 
 	metaBaseURL := ""
 	networkSource := false
+	hostname := ""
```

Comment:
this seems like a completely different way of setting hostnames, I'm not sure I cleary understand this change

Reply:
yep it can be just
```
var hostname string
```